src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PackageRestore/INuGetRestoreService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using Microsoft.VisualStudio.Composition;
+using Microsoft.VisualStudio.ProjectSystem.PackageRestore;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.PackageRestore;
+
+/// <summary>
+///     Responsible for handling NuGet package restore requests.
+/// </summary>
+[ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ExactlyOne)]
+internal interface INuGetRestoreService
+{
+    /// <summary>
+    ///     Requests that the <see cref="INuGetRestoreService"/> restore the project using the data in the <paramref name="restoreData"/>.
+    /// </summary>
+    /// <param name="restoreData">
+    ///     The data needed to run the restore information itself.
+    /// </param>
+    /// <param name="inputVersions">
+    ///     The <see cref="ConfiguredProject"/> version information corresponding to <paramref name="restoreData"/>.
+    ///     Implementations can use this to determine if there is pending work that could lead to a restore in
+    ///     the near future.
+    /// </param>
+    /// <param name="cancellationToken">
+    ///     A token indicating when the restore operation should be cancelled (e.g. due to the project being
+    ///     unloaded).
+    /// </param>
+    /// <returns>
+    ///     A <see cref="Task{TResult}"/> indicating when the work associated with the restore request, if any,
+    ///     has completed. The <see cref="Task{TResult}.Result"/> will be <see langword="true"/> if the restore
+    ///     ran and completed successfully, and <see langword="false"/> otherwise.
+    /// </returns>
+    public Task<bool> NominateAsync(ProjectRestoreInfo restoreData, IReadOnlyCollection<PackageRestoreConfiguredInput> inputVersions, CancellationToken cancellationToken);
```

Comment:
Remove `public` from interface member (ditto below).

Reply:
Now I'm scratching my head over why the compiler didn't complain about that...

Reply:
Ah, right.

Reply:
Done.